unimport/refactor.py

```diff
@@ -1,137 +1,111 @@
+        node_value = children[0]
+        while True:
+            if hasattr(node_value, "attr"):
+                module_attr.append(node_value.attr.value)
+                if hasattr(node_value, "value"):  # get parent
+                    node_value = node_value.value
+                else:
+                    module_attr.append(node_value.value.value)
+                    break
             else:
-                package_name = imp["module"].__name__
-                name_leafs = [
-                    Leaf(
-                        token.NAME,
-                        ", ".join(sorted(imp["modules"])),
-                        prefix=" ",
-                    ),
-                    Newline(),
-                ]
-                return FromImport(package_name, name_leafs)
-        return self.transform_inner_children(node, results["items"])
-
-    def import_name(self, node, results):
-        return self.transform_inner_children(node, results["imp"])
+                module_attr.append(node_value.value)
+                break
+        module_attr.reverse()
+        return ".".join(module_attr)
 
-    def get_imp_if_equal_to_lineno(self, lineno):
-        for imp in self.unused_modules:
-            if imp["lineno"] == lineno:
-                return imp
-
-    def transform_inner_children(self, node, imports):
-        if imports.type == syms.import_as_name or not imports.children:
-            children = [imports]
-        else:
-            children = imports.children
-        trailing_comma = None
-        if children[-1].type == token.COMMA:
-            # if end of children's char is equal to ',' then remove it
-            trailing_comma = children.pop()
-        commas = children[1:-1:2]
-        module_nodes = children[::2]
-        modules = tuple(traverse_imports(imports))
-        remove_counter = 0
-        for index, module in enumerate(modules):
-            if self.is_module_unused(module, node):
-                if commas:
-                    if index + 1 == len(modules):
-                        comma = commas.pop(index - remove_counter - 1)
-                        if trailing_comma:
-                            trailing_comma.remove()
-                            trailing_comma = None
-                    else:
-                        comma = commas.pop(index - remove_counter)
-                    comma.remove()
-                module_nodes.pop(index - remove_counter).remove()
-                remove_counter += 1
-        if remove_counter == len(modules):
-            return BlankLine()
-        if trailing_comma:
-            children.append(trailing_comma)
+    def is_import_used(self, import_name, location):
+        for imp in self.unused_imports:
+            if (
+                imp["name"] == import_name
+                and imp["lineno"] == location.start.line
+            ):
+                # unused
+                return False
+        return True
```

Comment:
I feel like you could use `all` here (or better just remove this and use all where you previously used this)

Reply:
I don't think I understand what you mean.